lib/restclient/request.rb

```diff
@@ -16,7 +16,10 @@ module RestClient
   # * :url
   # Optional parameters (have a look at ssl and/or uri for some explanations):
   # * :headers a hash containing the request headers
-  # * :cookies will replace possible cookies in the :headers
```

Comment:
I changed this because I don't think it's how the code works:

```ruby
@cookies = @headers.delete(:cookies) || args[:cookies] || {}
```

My preferred solution would actually be to invert this logic:

```ruby
@cookies = args[:cookies] || @headers.delete(:cookies) || {}
```

Or better yet, raise an error if _both_ `:cookies` AND `args[:headers][:cookies] values are passed.  What does the team think?